lib/ui/platform_dispatcher.dart

```diff
@@ -53,6 +53,16 @@ typedef _SetNeedsReportTimingsFunc = void Function(bool value);
 /// Signature for [PlatformDispatcher.onConfigurationChanged].
 typedef PlatformConfigurationChangedCallback = void Function(PlatformConfiguration configuration);
 
+/// Signature for [PlatformDispatcher.onError].
+///
+/// If this method returns false, the engine may use some fallback method to
+/// provide information about the error.
+///
+/// After calling this method, the process or the VM may terminate. Some severe
+/// unhandled errors may not be able to call this method either, such as Dart
+/// compilation errors or process terminating errors.
+typedef ErrorCallback = bool Function(Object exception, StackTrace? stackTrace);
```

Comment:
@hixie I've deviated from your design here - I think we just need a bool about whether the callback tried to handle the error or not. If an application wants to print anything, they can handle that themselves with `print` or `stderr.writeln` or whatever. Right now if this returns false, we retain existing behavior of writing the exception/stack to stderr (or logcat, or syslog, or whatever Fuchsia uses).